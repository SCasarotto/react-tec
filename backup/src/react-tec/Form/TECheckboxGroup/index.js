import TECheckboxGroup from './TECheckboxGroup'

export default TECheckboxGroup
