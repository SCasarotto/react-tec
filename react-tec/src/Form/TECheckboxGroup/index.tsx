import React, { ChangeEvent } from 'react'

import { TERowCustomProps } from '../TERow'
import TELabel, { TELabelCustomProps } from '../TELabel'
import { TECheckboxInputProps } from '../TECheckboxInput'

import {
	Container,
	InputWrapper,
	RowWrarpper,
	Checkbox,
	Label,
} from './styledComponents'

interface TECheckboxGroupButtonData {
	label: string
	value?: string
	key?: string
}
interface ManipulateRowData {
	rowData: string | TECheckboxGroupButtonData
	labelForKey: string
}
const manipulateRowData = ({
	rowData,
	labelForKey = '',
}: ManipulateRowData) => {
	//Allowing For Greater Shorthand
	if (typeof rowData === 'string') {
		const label = rowData
		const value = rowData
		const key = labelForKey + rowData
		return { label, value, key }
	} else if (typeof rowData === 'object') {
		let { label, value, key } = rowData

		if (!value) {
			value = label
		}
		if (!key) {
			key = labelForKey + label
		}

		return { label, value, key }
	}
	return rowData
}

interface TECheckboxGroupProps
	extends TERowCustomProps,
		TELabelCustomProps,
		Omit<TECheckboxInputProps, 'onChange'> {
	labelForKey: string
	onChange(values: string[], event: ChangeEvent<HTMLInputElement>): void
	inputRowSize?: string
	checkedValues: string[]
	buttonArray: (string | TECheckboxGroupButtonData)[]
	scrolling?: boolean
}
const TECheckboxGroup: React.FC<TECheckboxGroupProps> = (props) => {
	const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
		const { checkedValues, onChange } = props
		const { value } = event.target

		let newValues = [...checkedValues]
		if (checkedValues) {
			if (checkedValues.includes(value)) {
				newValues = newValues.filter((e) => e !== value)
			} else {
				newValues.push(value)
			}
		}
		if (onChange) {
			onChange(newValues, event)
		}
	}

	const {
		rowSize,
		last,
		className = '',
		title,
		required,
		inputRowSize,
		disabled,
		checkedValues,
		buttonArray,
		labelForKey,
	} = props

	return (
		<Container
			rowSize={rowSize}
			last={last}
			className={`TECheckboxGroup ${className}`}>
			<TELabel
				disabled={disabled}
				required={required}
				className="TECheckboxGroupTitle">
				{title}
			</TELabel>
			<InputWrapper className="TECheckboxGroupInputWrapper">
				{buttonArray &&
					buttonArray.map((rowData) => {
						const { label, value, key } = manipulateRowData({
							rowData,
							labelForKey,
						})

						return (
							<RowWrarpper
								key={key}
								rowSize={inputRowSize}
								className="TECheckboxGroupRowWrapper">
								<Checkbox
									value={value}
									onChange={handleInputChange}
									id={key}
									checked={
										checkedValues &&
										checkedValues.includes(value)
									}
									disabled={disabled}
									className="TECheckboxGroupCheckbox"
								/>
								<Label
									htmlFor={key}
									disabled={disabled}
									className="TECheckboxGroupLabel">
									{label}
								</Label>
							</RowWrarpper>
						)
					})}
			</InputWrapper>
		</Container>
	)
}

TECheckboxGroup.defaultProps = {
	scrolling: false,
}

export default TECheckboxGroup
