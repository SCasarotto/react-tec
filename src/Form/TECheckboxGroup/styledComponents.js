import styled from 'styled-components'

import TERow from './../TERow'
import TELabel from './../TELabel'
import TECheckboxInput from './../TECheckboxInput'

export const Container = styled(TERow)`
    ${(props) => {
        return `
            margin-bottom: 10px;
            display: inline-block;
            vertical-align: top;
        `
    }}
`
export const InputWrapper = styled.div`
    ${(props) => {
        const { scrolling, theme } = props
        return `
            display: flex;
            flex-wrap: wrap;
            ${
                scrolling
                    ? `
                    height: 300px;
                    overflow-y: scroll;
                    background-color: ${theme.white};
                    padding: 10px;
                    border: 1px solid ${theme.lightGray};
                    border-radius: 5;
                `
                    : ''
            }
        `
    }}
`

export const RowWrarpper = styled(TERow)`
    ${(props) => {
        return `
            margin-bottom: 5px;
            display: flex;
            align-items: center;
        `
    }}
`

export const Checkbox = styled(TECheckboxInput)`
    ${(props) => {
        return `
            display: inline-block;
            vertical-align: middle;
            cursor: pointer;
        `
    }}
`
export const Label = styled(TELabel)`
    ${(props) => {
        const { theme, disabled } = props

        return `
            width: auto;
            display: inline-block;
            vertical-align: middle;
            padding-left: 10px;
            margin-bottom: 0px;
            cursor: pointer;
            fontSize: 14px;
            color: ${disabled ? theme.gray : theme.darkerGray};

            @media (max-width: 550px) {
                fontSize: 12px;
            };
        `
    }}
`
