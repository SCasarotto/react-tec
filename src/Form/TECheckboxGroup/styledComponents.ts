import styled from 'styled-components'

import { TERow } from '../TERow'
import { TELabel } from '../TELabel'
import { TECheckboxInput } from '../TECheckboxInput'

export const Container = styled(TERow)`
	margin-bottom: 10px;
	display: inline-block;
	vertical-align: top;
`
export const InputWrapper = styled.div<{ scrolling?: boolean }>`
	display: flex;
	flex-wrap: wrap;
	height: ${(props) => (props.scrolling ? '300px' : 'auto')};
	overflow-y: ${(props) => (props.scrolling ? 'scroll' : 'visible')};
	background-color: ${(props) => (props.scrolling ? props.theme.white : 'transparent')};
	border: ${(props) => (props.scrolling ? `1px solid ${props.theme.lightGray}` : 'none')};
	padding: ${(props) => (props.scrolling ? '5px' : '0px')};
`

export const RowWrarpper = styled(TERow)`
	margin-bottom: 5px;
	display: flex;
	align-items: center;
`

export const Checkbox = styled(TECheckboxInput)`
	display: inline-block;
	vertical-align: middle;
	cursor: pointer;
`
export const Label = styled(TELabel)`
	width: auto;
	display: inline-block;
	vertical-align: middle;
	padding-left: 10px;
	margin-bottom: 0px;
	font-size: 14px;
	color: ${(props) => (props.disabled ? props.theme.gray : props.theme.darkerGray)};

	@media (max-width: 550px) {
		fontsize: 12px;
	}
`
