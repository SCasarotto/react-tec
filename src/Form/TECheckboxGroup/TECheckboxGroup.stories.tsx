import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TECheckboxGroup, TECheckboxGroupProps } from './TECheckboxGroup'

export default {
	title: 'Form/TECheckboxGroup',
	component: TECheckboxGroup,
	argTypes: {},
} as Meta

const Template: Story<TECheckboxGroupProps> = (args) => <TECheckboxGroup {...args} />

export const Default = Template.bind({})
Default.args = {
	labelForKey: 'TECheckboxGroup',
	title: 'TECheckboxGroup',
	buttonArray: ['one', '2', 'hour'],
}
