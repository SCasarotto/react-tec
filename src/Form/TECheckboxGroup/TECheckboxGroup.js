import React from 'react'
import PropTypes from 'prop-types'

import TELabel from './../TELabel'

import { Container, InputWrapper, RowWrarpper, Checkbox, Label } from './styledComponents'

const manipulateRowData = ({ rowData, labelForKey = '' }) => {
	//Allowing For Greater Shorthand
	if (typeof rowData === 'string') {
		const label = rowData
		const value = rowData
		const key = labelForKey + rowData
		return { label, value, key }
	} else if (typeof rowData === 'object') {
		let { label, value: value = label, key: key = labelForKey + label } = rowData
		return { label, value, key }
	}
}
const TECheckboxGroup = (props) => {
	const handleInputChange = (event) => {
		const { checkedValues, onChange } = props
		const { value } = event.target

		let newValues = [...checkedValues]
		if (checkedValues) {
			if (checkedValues.includes(value)) {
				newValues = newValues.filter((e) => e !== value)
			} else {
				newValues.push(value)
			}
		}
		onChange(newValues, event)
	}

	const {
		size,
		last,
		className = '',
		title,
		required,
		rowSize,
		disabled,
		checkedValues,
		buttonArray,
		labelForKey,
	} = props

	return (
		<Container size={size} last={last} className={`TECheckboxGroup ${className}`}>
			<TELabel disabled={disabled} required={required} className='TECheckboxGroupTitle'>
				{title}
			</TELabel>
			<InputWrapper className='TECheckboxGroupInputWrapper'>
				{buttonArray &&
					buttonArray.map((rowData) => {
						const { label, value, key } = manipulateRowData({
							rowData,
							labelForKey,
						})

						return (
							<RowWrarpper
								key={key}
								size={rowSize}
								className='TECheckboxGroupRowWrapper'
							>
								<Checkbox
									value={value}
									onChange={handleInputChange}
									id={key}
									checked={checkedValues && checkedValues.includes(value)}
									disabled={disabled}
									className='TECheckboxGroupCheckbox'
								/>
								<Label
									htmlFor={key}
									disabled={disabled}
									className='TECheckboxGroupLabel'
								>
									{label}
								</Label>
							</RowWrarpper>
						)
					})}
			</InputWrapper>
		</Container>
	)
}

TECheckboxGroup.propTypes = {
	onChange: PropTypes.func.isRequired,
	scrolling: PropTypes.bool,
	buttonArray: PropTypes.array.isRequired,
	checkedValues: PropTypes.array.isRequired,
	labelForKey: PropTypes.string.isRequired,
}

TECheckboxGroup.defaultProps = {
	scrolling: false,
}

export default TECheckboxGroup
