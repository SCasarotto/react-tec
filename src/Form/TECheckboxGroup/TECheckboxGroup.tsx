import React, { ChangeEvent } from 'react'

import { TERowCustomProps, TERowProps } from '../TERow'
import { TELabel, TELabelCustomProps } from '../TELabel'
import { TECheckboxInputProps } from '../TECheckboxInput'
import { manipulateRowData } from './../../helpers'

import { Container, InputWrapper, RowWrarpper, Checkbox, Label } from './styledComponents'

export interface TECheckboxGroupButtonData {
	label: string
	value?: string
	key?: string
}

export interface TECheckboxGroupProps
	extends TERowCustomProps,
		TELabelCustomProps,
		Omit<TECheckboxInputProps, 'onChange'> {
	labelForKey: string
	onChange(values: string[], event: ChangeEvent<HTMLInputElement>): void
	inputRowSize?: TERowProps['rowSize']
	checkedValues: string[]
	buttonArray: (string | TECheckboxGroupButtonData)[]
	scrolling?: boolean
}
export const TECheckboxGroup: React.FC<TECheckboxGroupProps> = (props) => {
	const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
		const { checkedValues, onChange } = props
		const { value } = event.target

		let newValues = [...checkedValues]
		if (checkedValues) {
			if (checkedValues.includes(value)) {
				newValues = newValues.filter((e) => e !== value)
			} else {
				newValues.push(value)
			}
		}
		if (onChange) {
			onChange(newValues, event)
		}
	}

	const {
		rowSize,
		last,
		className = '',
		title,
		required,
		inputRowSize,
		disabled,
		checkedValues,
		buttonArray,
		labelForKey,
	} = props

	return (
		<Container rowSize={rowSize} last={last} className={`TECheckboxGroup ${className}`}>
			{title && (
				<TELabel disabled={disabled} required={required} className='TECheckboxGroupTitle'>
					{title}
				</TELabel>
			)}
			<InputWrapper className='TECheckboxGroupInputWrapper'>
				{buttonArray &&
					buttonArray.map((rowData) => {
						const { label, value, key } = manipulateRowData({
							rowData,
							labelForKey,
						})

						return (
							<RowWrarpper
								key={key}
								rowSize={inputRowSize}
								className='TECheckboxGroupRowWrapper'
							>
								<Checkbox
									value={value}
									onChange={handleInputChange}
									id={key}
									checked={checkedValues && checkedValues.includes(value)}
									disabled={disabled}
									className='TECheckboxGroupCheckbox'
								/>
								<Label
									htmlFor={key}
									disabled={disabled}
									className='TECheckboxGroupLabel'
								>
									{label}
								</Label>
							</RowWrarpper>
						)
					})}
			</InputWrapper>
		</Container>
	)
}

TECheckboxGroup.defaultProps = {
	scrolling: false,
}
