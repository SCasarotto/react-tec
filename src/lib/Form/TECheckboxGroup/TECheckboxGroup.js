//
//TE Version 0.3.0
//

import React from 'react'
import PropTypes from 'prop-types'

import TELabel from './../TELabel'

import { Container, InputWrapper, RowWrarpper, Checkbox, Label } from './styledComponents'

const manipulateRowData = ({ rowData, labelForKey = '' }) => {
    //Allowing For Greater Shorthand
    if (typeof rowData === 'string') {
        const label = rowData
        const value = rowData
        const key = labelForKey + rowData
        return { label, value, key }
    } else if (typeof rowData === 'object') {
        let { label, value: value = label, key: key = labelForKey + label } = rowData
        return { label, value, key }
    }
}
const TECheckboxGroup = (props) => {
    const handleInputChange = (event) => {
        const { checkedValues, onChange } = props
        const { value, id } = event.target

        let newValues = [...checkedValues]
        if (checkedValues) {
            if (checkedValues.includes(value)) {
                newValues = newValues.filter((e) => e !== value)
            } else {
                newValues.push(value)
            }
        }
        onChange(newValues, id)
    }

    const {
        size,
        last,
        title,
        required,
        rowSize,
        disabled,
        checkedValues,
        buttonArray,
        labelForKey,
    } = props

    return (
        <Container size={size} last={last}>
            <TELabel disabled={disabled} required={required}>
                {title}
            </TELabel>
            <InputWrapper>
                {buttonArray &&
                    buttonArray.map((rowData) => {
                        const { label, value, key } = manipulateRowData({
                            rowData,
                            labelForKey,
                        })

                        return (
                            <RowWrarpper key={key} size={rowSize}>
                                <Checkbox
                                    value={value}
                                    onChange={handleInputChange}
                                    id={key}
                                    checked={checkedValues && checkedValues.includes(value)}
                                    disabled={disabled}
                                />
                                <Label htmlFor={key} disabled={disabled}>
                                    {label}
                                </Label>
                            </RowWrarpper>
                        )
                    })}
            </InputWrapper>
        </Container>
    )
}

TECheckboxGroup.propTypes = {
    scrolling: PropTypes.bool,
    buttonArray: PropTypes.array.isRequired,
    checkedValues: PropTypes.array.isRequired,
    labelForKey: PropTypes.string,
}

TECheckboxGroup.defaultProps = {
    scrolling: false,
}

export default TECheckboxGroup
