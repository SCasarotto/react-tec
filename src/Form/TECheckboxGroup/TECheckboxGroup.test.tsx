import React from 'react'
import { render } from '@testing-library/react'
import { TECheckboxGroup } from './TECheckboxGroup'

describe('TECheckboxGroup', () => {
	// Renders Without Error
	it('matches snapshot', () => {
		const { asFragment } = render(
			<TECheckboxGroup
				labelForKey='testTECheckboxGroup'
				onChange={() => {}}
				buttonArray={['someValue', 'AnotherValue']}
				checkedValues={['someValue']}
			/>,
		)
		expect(asFragment()).toMatchSnapshot()
	})

	// Class Name
	it('accepts custom className', () => {
		const { container } = render(
			<TECheckboxGroup
				className='myClassName'
				labelForKey='testTECheckboxGroup'
				onChange={() => {}}
				buttonArray={['someValue', 'AnotherValue']}
				checkedValues={['someValue']}
			/>,
		)
		expect(container.firstChild).toHaveClass('myClassName')
	})
})
