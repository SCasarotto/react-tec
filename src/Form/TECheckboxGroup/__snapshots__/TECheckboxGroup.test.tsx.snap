// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TECheckboxGroup matches snapshot 1`] = `
<DocumentFragment>
  .c6 {
  position: relative;
  display: block;
  width: auto;
  box-sizing: border-box;
  margin-bottom: 3px;
  font-size: 18px;
  cursor: pointer;
}

.c0 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  margin-bottom: 10px;
  text-align: left;
  box-sizing: border-box;
  padding-right: 0px;
  width: 100%;
}

.c4 {
  cursor: pointer;
  box-sizing: border-box;
  margin: 0px;
}

.c4:disabled {
  cursor: not-allowed;
}

.c1 {
  margin-bottom: 10px;
  display: inline-block;
  vertical-align: top;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  height: auto;
  overflow-y: visible;
  background-color: transparent;
  border: none;
  padding: 0px;
}

.c3 {
  margin-bottom: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  display: inline-block;
  vertical-align: middle;
  cursor: pointer;
}

.c7 {
  width: auto;
  display: inline-block;
  vertical-align: middle;
  padding-left: 10px;
  margin-bottom: 0px;
  font-size: 14px;
  cursor: pointer;
}

@media (max-width:550px) {
  .c6 {
    font-size: 16px;
  }
}

@media (max-width:550px) {
  .c7 {
    fontsize: 12px;
  }
}

<div
    class="c0 TERow c1 TECheckboxGroup "
  >
    <div
      class="c2 TECheckboxGroupInputWrapper"
    >
      <div
        class="c0 TERow c3 TECheckboxGroupRowWrapper"
      >
        <input
          checked=""
          class="c4 TECheckboxInput c5 TECheckboxGroupCheckbox"
          id="testTECheckboxGroupsomeValue"
          type="checkbox"
          value="someValue"
        />
        <label
          class="c6 TELabel c7 TECheckboxGroupLabel"
          for="testTECheckboxGroupsomeValue"
        >
          someValue
        </label>
      </div>
      <div
        class="c0 TERow c3 TECheckboxGroupRowWrapper"
      >
        <input
          class="c4 TECheckboxInput c5 TECheckboxGroupCheckbox"
          id="testTECheckboxGroupAnotherValue"
          type="checkbox"
          value="AnotherValue"
        />
        <label
          class="c6 TELabel c7 TECheckboxGroupLabel"
          for="testTECheckboxGroupAnotherValue"
        >
          AnotherValue
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
